commit 3559d72d8a1517f5042667b1e7224be4225311fb (refs/changes/38/438/2)
Author: David Pacheco <dap@joyent.com>
Date:   2016-09-08T15:27:54-07:00 (3 years, 1 month ago)
    
    MANTA-2981 msplit reports ConnectTimeoutError with large objects, many reducers
    MANTA-2982 manta-compute-bin should use semver dep on node-manta

diff --git a/bin/msplit b/bin/msplit
index c915874..c8ad497 100755
--- a/bin/msplit
+++ b/bin/msplit
@@ -1,6 +1,24 @@
 #!/usr/bin/env node
 // -*- mode: js -*-
-// Copyright (c) 2013, Joyent, Inc. All rights reserved.
+// Copyright (c) 2016, Joyent, Inc. All rights reserved.
+
+/*
+ * The number of concurrent upstream connections should be less than the maximum
+ * number that Manta allows from a single task before queueing them.  This is
+ * part of Marlin's configuration, but it's not exposed to user tasks, so we
+ * hardcode the default number here.  A better approach would be to have the
+ * server issue a 429 "Too Many Requests" response (instead of queueing them)
+ * and have the client back off when this happens.
+ */
+var msConcurrency = 25;
+
+/*
+ * Configure maxSockets based on our desired concurrency.  We have to do this
+ * here, before pulling in "restify-clients" (via "manta"), because
+ * "restify-clients" reads these from the top-level.
+ */
+var mod_http = require('http');
+mod_http.globalAgent.maxSockets = msConcurrency;
 
 var mod_assert = require('assert');
 var mod_bunyan = require('bunyan');
@@ -85,16 +103,6 @@ var msObjPrefix = null;
 var msStreams = [];
 var msTmpFilePrefix = '/var/tmp/msplit.' + process.pid + '.';
 
-/*
- * The number of concurrent upstream connections should be less than the maximum
- * number that Manta allows from a single task before queueing them.  This is
- * part of Marlin's configuration, but it's not exposed to user tasks, so we
- * hardcode the default number here.  A better approach would be to have the
- * server issue a 429 "Too Many Requests" response (instead of queueing them)
- * and have the client back off when this happens.
- */
-var msConcurrency = 25;
-
 function main()
 {
 	var lstream;
diff --git a/package.json b/package.json
index 1b10ce9..f158a5c 100644
--- a/package.json
+++ b/package.json
@@ -9,7 +9,7 @@
         "assert-plus": "0.1.2",
         "bunyan": "0.20.0",
         "lstream": "0.0.4",
-        "manta": "git+ssh://git@github.com:joyent/node-manta.git#master",
+        "manta": "^3.0.0",
         "memorystream": "0.2.0",
         "node-uuid": "1.4.0",
         "posix-getopt": "1.0.0",
